tests/test_aiocassandra.py

```diff
@@ -50,6 +63,94 @@
         await fut
 
 
+@pytest.mark.asyncio
+async def test_execute_future_cancel(cassandra, caplog, loop):
+    cql = 'SELECT now() as now FROM system.local;'
+
+    old_fut_factory = cassandra._asyncio_fut_factory
+
+    def new_patch_factory():
+        fut = old_fut_factory()
+        fut.cancel()
+        return fut
+
+    cassandra._asyncio_fut_factory = new_patch_factory
+
+    fut = cassandra.execute_future(cql)
+
+    assert asyncio.isfuture(fut)
+
+    with caplog.at_level(logging.ERROR):
+        await asyncio.sleep(0.1, loop=loop)
+        assert len(caplog.records) == 0
+
+    with pytest.raises(asyncio.CancelledError):
+        await fut
+
+
+@pytest.mark.asyncio
+async def test_execute_future_cancel_error(cassandra, caplog, loop):
+    cql = 'SELECT 1;'
+
+    old_fut_factory = cassandra._asyncio_fut_factory
+
+    def new_patch_factory():
+        fut = old_fut_factory()
+        fut.cancel()
+        return fut
+
+    cassandra._asyncio_fut_factory = new_patch_factory
+
+    fut = cassandra.execute_future(cql)
+
+    assert asyncio.isfuture(fut)
+
+    with caplog.at_level(logging.ERROR):
+        await asyncio.sleep(0.1, loop=loop)
+        assert len(caplog.records) == 0
+
+    with pytest.raises(asyncio.CancelledError):
+        await fut
+
+
+@pytest.mark.asyncio
+async def test_execute_futures_simple(cassandra):
+    cql = 'SELECT now() as now FROM system.local;'
+
+    ret = []
+
+    async for row in cassandra.execute_futures(cql):
+        ret.append(row)
+
+    assert len(ret) == 1
+
+    assert isinstance(ret[0].now, uuid.UUID)
+
+
+@pytest.mark.asyncio
+async def test_execute_futures_simple_statement_empty(cassandra):
+    cql = 'SELECT * FROM system_schema.types;'
+    statement = SimpleStatement(cql, fetch_size=1)
+
+    async for _ in cassandra.execute_futures(statement):
```

Comment:
this piece of code is fairly unobvious and deserves to have comments

Reply:
just populate list and test it for empty